dbt_coves/tasks/generate/sources.py

```diff
@@ -222,9 +223,22 @@ def generate_sources(self, rels):
 
     def generate_model(self, relation, destination, options):
         destination.parent.mkdir(parents=True, exist_ok=True)
-        columns = self.adapter.get_columns_in_relation(relation)
-        nested_field_type = NESTED_FIELD_TYPES.get(self.adapter.__class__.__name__)
-        nested = [col.name.lower() for col in columns if col.dtype == nested_field_type]
+        metadata = self.get_metadata(self.get_config_value("metadata"))
```

Comment:
@LeonardoPelanda the issue here is that metadata is loaded each time that `generate_model` is called. It is called multiple times. The metadata should be loaded at the beginning and passed on.